src/adapter/src/catalog.rs

```diff
@@ -2853,14 +2892,9 @@ impl Catalog {
                                 index.name, e
                             )
                         });
-                    let CatalogItem::Index(catalog_index) = &mut item else {
+                    let CatalogItem::Index(_) = &mut item else {
                         panic!("internal error: builtin index {}'s SQL does not begin with \"CREATE INDEX\".", index.name);
                     };
-                    if index.is_retained_metrics_object {
-                        catalog_index.is_retained_metrics_object = true;
-                        catalog_index.custom_logical_compaction_window =
-                            Some(catalog.state.system_config().metrics_retention());
-                    }
```

Comment:
Why does this get removed?

Reply:
Because these are now set by the `parse_item` function, which is called above.